src/wp-includes/blocks/index.php

```diff
@@ -12,6 +12,93 @@
 require BLOCKS_PATH . 'widget-group.php';
 require BLOCKS_PATH . 'require-dynamic-blocks.php';
 
+/**
+ * Registers core block style handles.
+ *
+ * While {@see register_block_style_handle()} is typically used for that, the way it is
+ * implemented is inefficient for core block styles. Registering those style handles here
+ * avoids unnecessary logic and filesystem lookups in the other function.
+ *
+ * @since 6.3.0
+ */
+function register_core_block_style_handles() {
+	if ( ! wp_should_load_separate_core_block_assets() ) {
+		return;
+	}
+
+	static $core_blocks_meta;
+	if ( ! $core_blocks_meta ) {
+		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
+	}
+
+	$includes_url  = includes_url();
+	$includes_path = ABSPATH . WPINC . '/';
+	$suffix        = wp_scripts_get_suffix();
+	$wp_styles     = wp_styles();
+	$style_fields  = array(
+		'style'       => 'style',
+		'editorStyle' => 'editor',
+	);
+	$files         = array();
+
+	/*
+	 * Ignore transient cache when `WP_DEBUG` is enabled. Why? To avoid interfering with
+	 * the core developer's workflow.
+	 *
+	 * @todo Replace `WP_DEBUG` once an "in development mode" check is available in Core.
+	 */
+	if ( ! WP_DEBUG ) {
+		$transient_name = 'wp_core_block_styles';
+		$files          = get_transient( $transient_name );
+		if ( ! $files ) {
+			$files = glob( __DIR__ . "/**/**.css" );
+			set_transient( $transient_name, $files );
+		}
+	}
```

Comment:
I think you have a bug here now, if `WP_DEBUG` is enabled, `$files` will always be an empty array.
```suggestion

	/*
	 * Ignore transient cache when `WP_DEBUG` is enabled. Why? To avoid interfering with
	 * the core developer's workflow.
	 *
	 * @todo Replace `WP_DEBUG` once an "in development mode" check is available in Core.
	 */
	if ( ! WP_DEBUG ) {
		$transient_name = 'wp_core_block_styles';
		$files          = get_transient( $transient_name );
		if ( ! $files ) {
			$files = glob( __DIR__ . "/**/**.css" );
			set_transient( $transient_name, $files );
		}
	} else {
		$files = glob( __DIR__ . "/**/**.css" );
	}
```